presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/TimestampStatistics.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata.statistics;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class TimestampStatistics
+        implements RangeStatistics<Long>
+{
+    // 1 byte to denote if null + 4 bytes for the value (date is of integer type)
+    public static final long TIMESTAMP_VALUE_BYTES = Byte.BYTES + Long.BYTES;
+
+    private final Long minimum;
+    private final Long maximum;
+
+    public TimestampStatistics(Long minimum, Long maximum)
+    {
+        checkArgument(minimum == null || maximum == null || minimum <= maximum, "minimum is not less than maximum");
```

Comment:
Is `minimum == null || maximum == null` an open end? Why not use `Long.MIN/AX_VALUE`?

btw, if this `TimestampStatistics` were always built with `TimestampStatisticsBuilder`, nulls don't seem possible to happen here.

Reply:
`TimestampStatistics` is not always built with `TimestampStatisticsBuilder`, in case of `OrcMetadataReader` or `DwrfMetadataReader` we are directly populating `TimestampStatistics` and  there could be null in the min and max value so adding the check.